pkg/logs/internal/decoder/multiline_handler.go

```diff
@@ -163,7 +163,12 @@ func (h *MultiLineHandler) sendBuffer() {
 				telemetry.GetStatsTelemetryProvider().Count(linesCombinedTelemetryMetricName, float64(linesCombined), []string{})
 			}
 		}
-		msg := message.NewRawMessage(content, h.status, h.linesLen, h.timestamp)
+		msg := h.msg
+		msg.SetContent(content)
+		msg.RawDataLen = h.linesLen
+		msg.Status = h.msg.Status
```

Comment:
I think we can omit this line since `msg` already retains the correct status, and the next line `msg.ParsingExtra.Timestamp = h.timestamp` since `msg` already contains the original timestamp. 
I think we can also remove the `h.timestamp` field from the struct. 

Otherwise, LGTM